### PR TITLE
[MIP] Remove ProblemFiller::updateBetweenSensiIteration() method

### LINKED ISSUE #1049: Improve MIP code
### Describe the current behavior

In https://github.com/powsybl/powsybl-open-rao/pull/1014, we had to do a workaround in order to fix a bug, by filtering NaN values from the MIP. In order to go fast, we had to replace the "updateBetweenSensiIterations" content by a reset of the matrix, followed by a "fill" and an old "update". The "update" is only needed because its signature allows passing more info to the matrix than "fill".

### Describe the expected behavior

Make the "fill" methods more complete, and remove the need of the "updateBetweenSensiIterations" methods in the fillers. This will make the code more simple.

### Describe the motivation

Cleaner code

### Extra Information

_No response_

### PR DESCRIPTION
**Pull request by pet-mit**
**Please check if the PR fulfills these requirements**

- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem?**

Fixes #1049

### CODE REVIEW

File: ra-optimisation/search-tree-rao/src/main/java/com/powsybl/openrao/searchtreerao/linearoptimisation/algorithms/fillers/MaxLoopFlowFiller.java [hunk 1]
```diff
@@ -50,13 +51,13 @@ private Set<FlowCnec> getValidLoopFlowCnecs(SensitivityResult sensitivityResult)
...
     @Override
-    public void fill(LinearProblem linearProblem, FlowResult flowResult, SensitivityResult sensitivityResult) {
-        buildLoopFlowConstraintsAndUpdateObjectiveFunction(linearProblem, getValidLoopFlowCnecs(sensitivityResult), flowResult);
-    }
-
-    @Override
-    public void updateBetweenSensiIteration(LinearProblem linearProblem, FlowResult flowResult, SensitivityResult sensitivityResult, RangeActionActivationResult rangeActionActivationResult) {
-        updateLoopFlowConstraints(linearProblem, getValidLoopFlowCnecs(sensitivityResult), flowResult);
+    public void fill(LinearProblem linearProblem, FlowResult flowResult, SensitivityResult sensitivityResult, RangeActionActivationResult rangeActionActivationResult) {
+        // TODO : we may argue that "ptdfApproximationLevel.shouldUpdatePtdfWithPstChange()=false" has no real advantage anymore
```
Comment (phiedw) [inline ra-optimisation/search-tree-rao/src/main/java/com/powsybl/openrao/searchtreerao/linearoptimisation/algorithms/fillers/MaxLoopFlowFiller.java:55]: We can still avoid computing the zone to zone ptdfs (is the sensitivities of the glsks on the cnecs) during the sensitivity computation with the parameter set to false.

File: ra-optimisation/search-tree-rao/src/main/java/com/powsybl/openrao/searchtreerao/linearoptimisation/algorithms/fillers/MaxMinRelativeMarginFiller.java [hunk 2]
```diff
@@ -52,25 +52,18 @@ public MaxMinRelativeMarginFiller(Set<FlowCnec> optimizedCnecs,
...
-    public void fill(LinearProblem linearProblem, FlowResult flowResult, SensitivityResult sensitivityResult) {
-        super.fill(linearProblem, flowResult, sensitivityResult);
+    public void fill(LinearProblem linearProblem, FlowResult flowResult, SensitivityResult sensitivityResult, RangeActionActivationResult rangeActionActivationResult) {
+        super.fill(linearProblem, flowResult, sensitivityResult, rangeActionActivationResult);
         buildMinimumRelativeMarginSignBinaryVariable(linearProblem);
         updateMinimumNegativeMarginDefinition(linearProblem);
         Set<FlowCnec> validFlowCnecs = FillersUtil.getFlowCnecsComputationStatusOk(optimizedCnecs, sensitivityResult);
         buildMinimumRelativeMarginVariable(linearProblem, validFlowCnecs);
-        buildMinimumRelativeMarginConstraints(linearProblem, validFlowCnecs);
+        // TODO : we may argue that "ptdfApproximationLevel.shouldUpdatePtdfWithPstChange()=false" has no real advantage anymore
```
Comment (phiedw) [inline ra-optimisation/search-tree-rao/src/main/java/com/powsybl/openrao/searchtreerao/linearoptimisation/algorithms/fillers/MaxMinRelativeMarginFiller.java:61]: We can still avoid computing the zone to zone ptdfs (is the sensitivities of the glsks on the cnecs) during the sensitivity computation with the parameter set to false.

File: ra-optimisation/search-tree-rao/src/main/java/com/powsybl/openrao/searchtreerao/linearoptimisation/algorithms/linearproblem/LinearProblem.java [hunk 3]
```diff
@@ -81,9 +82,10 @@ public static LinearProblemBuilder create() {
         return new LinearProblemBuilder();
     }
 
-    LinearProblem(List<ProblemFiller> fillerList, RangeActionsOptimizationParameters.Solver solver, double relativeMipGap, String solverSpecificParameters) {
+    LinearProblem(List<ProblemFiller> fillerList, RangeActionActivationResult raActivationFromParentLeaf, RangeActionsOptimizationParameters.Solver solver, double relativeMipGap, String solverSpecificParameters) {
```
Comment (phiedw) [inline ra-optimisation/search-tree-rao/src/main/java/com/powsybl/openrao/searchtreerao/linearoptimisation/algorithms/linearproblem/LinearProblem.java:85]: what happens in the cases where the activation is reset from one leaf to the next?

Comment (pet-mit) [reply ra-optimisation/search-tree-rao/src/main/java/com/powsybl/openrao/searchtreerao/linearoptimisation/algorithms/linearproblem/LinearProblem.java:85]: I'm not sure I understand your question ; but for me the new version is strictly equivalent to the old one (see changes in LinearProblem, LinearProblemBuilder, and fillers)

**Review by phiedw — APPROVED** [review]

**Pull request merged**